### PR TITLE
Run strip down version without ros

### PR DESCRIPTION
**Pull request by jviereck**
This PR implements a way to run a minimal subset of DGM without ros & dynamic graph. This is needed for running on the raspberry pi, where ros is not available.

In detail, this PR implements a new option `BUILD_ROS_DYNAMIC_GRAPH` which is ON by default. When the setting is off, only the hardware process and the dynamic graph head code is built. All other code (ros integration + dynamic graph integration) are disabled.

### CODE REVIEW

File: include/dynamic_graph_manager/tools.hpp [hunk 1]
```diff
@@ -7,13 +7,22 @@
...
  */
 
-#include <dynamic-graph/linear-algebra.h>
-
 #include "yaml_utils/yaml_cpp_fwd.hpp"
 
 #ifndef TOOLS_HH
 #define TOOLS_HH
 
+#pragma GCC diagnostic push
```
Comment (MaximilienNaveau) [inline include/dynamic_graph_manager/tools.hpp:15]: dangerous for real-time?

File: CMakeLists.txt [hunk 2]
```diff
@@ -88,15 +103,17 @@ add_subdirectory(src)
 #
 add_subdirectory(srcpy)
 
-#
-# manage the demos
-#
-add_subdirectory(demos)
+if (BUILD_ROS_DYNAMIC_GRAPH)
```
Comment (MaximilienNaveau) [inline CMakeLists.txt]: the if  should be inside the `demos/CMakeLists.txt` here

File: CMakeLists.txt [hunk 3]
```diff
@@ -88,15 +103,17 @@ add_subdirectory(src)
...
+    # manage the demos
+    #
+    add_subdirectory(demos)
+endif()
 
 #
 # manage the unit tests
 #
-if(BUILD_TESTING)
+if (BUILD_ROS_DYNAMIC_GRAPH AND BUILD_TESTING)
```
Comment (MaximilienNaveau) [inline CMakeLists.txt]: the if  should be inside the `tests/CMakeLists.txt` here

**Review by MaximilienNaveau — CHANGES_REQUESTED** [review]
Some changes in the CMake but overall it's good, thanks.

File: include/dynamic_graph_manager/tools.hpp (hunk 1 above)
Comment (jviereck) [reply include/dynamic_graph_manager/tools.hpp:15]: Why do you think this line is dangerous for real-time? I literately copied the code from `dynamic-graph/linear-algebra.h` [1] and they use the same analysis. So if there is a problem with real-time compatibility, we would have it already right now.

[1]: http://docs.ros.org/en/melodic/api/dynamic-graph/html/linear-algebra_8h_source.html

File: CMakeLists.txt (hunk 2 above)
Comment (jviereck) [reply CMakeLists.txt]: I addressed this one.

File: CMakeLists.txt (hunk 3 above)
Comment (jviereck) [reply CMakeLists.txt]: I addressed this one.

**Pull request merged**